dynamicExecutable/CMakeLists.txt

```diff
@@ -1,32 +1,54 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.15)
 project(shastaDynamicExecutable)
 
+# Use nvcc for GPU compilation
+if(BUILD_FOR_GPU)
+    set(CMAKE_CUDA_COMPILER /usr/local/cuda/bin/nvcc)
+    find_package(CUDA 10.0 REQUIRED)
+    set(CMAKE_CUDA_COMPILER_ENV_VAR "CUDACXX")
+    set(CMAKE_CUDA_HOST_COMPILER_ENV_VAR "CUDAHOSTCXX")
+    project(shastaDynamicExecutable LANGUAGES C CXX CUDA)
+endif(BUILD_FOR_GPU)
+
 # C++ dialect.
-add_definitions(-std=c++14)
+#add_definitions(-std=c++14)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
```

Comment:
I find `add_definitions` more readable than `set(CMAKE_CXX_FLAGS`, and [cmake documentation](https://cmake.org/cmake/help/v3.0/command/add_definitions.html) says that `add_definitions` can be used to add any flag, not just `-D` definitions.  Is there a reason why you were forced to switch to `set(CMAKE_CXX_FLAGS`?

Reply:
Yes, since we're also using NVCC compiler for GPU code, add_definitions would set flags for both compilers and NVCC would throw an error for these.

Reply:
In that case, if possible it would be better to put all of the GPU code in a separate directory, and create a separate cmake file for that. It's not good that the regular cmake files have to worry about the GPU compiler.

Reply:
That might be possible - I can try doing that.

Reply:
Please give it a try. I don't want to create a situation where cmake changes for the CPU code break the GPU build or vice versa.